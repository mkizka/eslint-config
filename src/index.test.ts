import { ESLint } from "eslint";
import { describe, expect, test } from "vitest";

describe("スナップショットテスト", () => {
  test("index.js", async () => {
    const eslint = new ESLint();
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    const result = await eslint.calculateConfigForFile("index.js");
    expect(result).toMatchSnapshot();
  });
  test("index.ts", async () => {
    const eslint = new ESLint();
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    const result = await eslint.calculateConfigForFile("index.ts");
    expect(result).toMatchSnapshot();
  });
});
