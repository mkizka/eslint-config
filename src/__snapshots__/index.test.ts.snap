// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`スナップショットテスト > index.js 1`] = `
{
  "language": "@/js",
  "languageOptions": {
    "ecmaVersion": 2025,
    "globals": {
      "AbortController": false,
      "AbortSignal": false,
      "Blob": false,
      "BroadcastChannel": false,
      "Buffer": false,
      "ByteLengthQueuingStrategy": false,
      "CloseEvent": false,
      "CompressionStream": false,
      "CountQueuingStrategy": false,
      "Crypto": false,
      "CryptoKey": false,
      "CustomEvent": false,
      "DOMException": false,
      "DecompressionStream": false,
      "Event": false,
      "EventTarget": false,
      "File": false,
      "FormData": false,
      "Headers": false,
      "MessageChannel": false,
      "MessageEvent": false,
      "MessagePort": false,
      "Navigator": false,
      "Performance": false,
      "PerformanceEntry": false,
      "PerformanceMark": false,
      "PerformanceMeasure": false,
      "PerformanceObserver": false,
      "PerformanceObserverEntryList": false,
      "PerformanceResourceTiming": false,
      "ReadableByteStreamController": false,
      "ReadableStream": false,
      "ReadableStreamBYOBReader": false,
      "ReadableStreamBYOBRequest": false,
      "ReadableStreamDefaultController": false,
      "ReadableStreamDefaultReader": false,
      "Request": false,
      "Response": false,
      "SubtleCrypto": false,
      "TextDecoder": false,
      "TextDecoderStream": false,
      "TextEncoder": false,
      "TextEncoderStream": false,
      "TransformStream": false,
      "TransformStreamDefaultController": false,
      "URL": false,
      "URLSearchParams": false,
      "WebAssembly": false,
      "WebSocket": false,
      "WritableStream": false,
      "WritableStreamDefaultController": false,
      "WritableStreamDefaultWriter": false,
      "__dirname": false,
      "__filename": false,
      "atob": false,
      "btoa": false,
      "clearImmediate": false,
      "clearInterval": false,
      "clearTimeout": false,
      "console": false,
      "crypto": false,
      "exports": true,
      "fetch": false,
      "global": false,
      "module": false,
      "navigator": false,
      "performance": false,
      "process": false,
      "queueMicrotask": false,
      "require": false,
      "setImmediate": false,
      "setInterval": false,
      "setTimeout": false,
      "structuredClone": false,
    },
    "parser": "espree@10.3.0",
    "parserOptions": {
      "ecmaFeatures": {
        "jsx": true,
      },
      "jsxPragma": null,
      "sourceType": "module",
    },
    "sourceType": "module",
  },
  "linterOptions": {
    "reportUnusedDisableDirectives": 2,
  },
  "plugins": [
    "@",
    "simple-import-sort:eslint-plugin-simple-import-sort@12.1.1",
    "unused-imports:unused-imports",
    "eqeqeq-fix",
    "react",
    "react-hooks:eslint-plugin-react-hooks",
  ],
  "processor": undefined,
  "rules": {
    "@typescript-eslint/eqeqeq": [
      0,
    ],
    "@typescript-eslint/no-unused-vars": [
      0,
    ],
    "constructor-super": [
      2,
    ],
    "eqeqeq": [
      0,
    ],
    "eqeqeq-fix/eqeqeq": [
      2,
    ],
    "for-direction": [
      2,
    ],
    "getter-return": [
      2,
      {
        "allowImplicit": false,
      },
    ],
    "no-async-promise-executor": [
      2,
    ],
    "no-case-declarations": [
      2,
    ],
    "no-class-assign": [
      2,
    ],
    "no-compare-neg-zero": [
      2,
    ],
    "no-cond-assign": [
      2,
      "except-parens",
    ],
    "no-console": [
      2,
      {},
    ],
    "no-const-assign": [
      2,
    ],
    "no-constant-binary-expression": [
      2,
    ],
    "no-constant-condition": [
      2,
      {
        "checkLoops": "allExceptWhileTrue",
      },
    ],
    "no-control-regex": [
      2,
    ],
    "no-debugger": [
      2,
    ],
    "no-delete-var": [
      2,
    ],
    "no-dupe-args": [
      2,
    ],
    "no-dupe-class-members": [
      2,
    ],
    "no-dupe-else-if": [
      2,
    ],
    "no-dupe-keys": [
      2,
    ],
    "no-duplicate-case": [
      2,
    ],
    "no-empty": [
      2,
      {
        "allowEmptyCatch": false,
      },
    ],
    "no-empty-character-class": [
      2,
    ],
    "no-empty-pattern": [
      2,
      {
        "allowObjectPatternsAsParameters": false,
      },
    ],
    "no-empty-static-block": [
      2,
    ],
    "no-ex-assign": [
      2,
    ],
    "no-extra-boolean-cast": [
      2,
      {},
    ],
    "no-fallthrough": [
      2,
      {
        "allowEmptyCase": false,
        "reportUnusedFallthroughComment": false,
      },
    ],
    "no-func-assign": [
      2,
    ],
    "no-global-assign": [
      2,
      {
        "exceptions": [],
      },
    ],
    "no-import-assign": [
      2,
    ],
    "no-invalid-regexp": [
      2,
      {},
    ],
    "no-irregular-whitespace": [
      2,
      {
        "skipComments": false,
        "skipJSXText": false,
        "skipRegExps": false,
        "skipStrings": true,
        "skipTemplates": false,
      },
    ],
    "no-loss-of-precision": [
      2,
    ],
    "no-misleading-character-class": [
      2,
    ],
    "no-new-native-nonconstructor": [
      2,
    ],
    "no-nonoctal-decimal-escape": [
      2,
    ],
    "no-obj-calls": [
      2,
    ],
    "no-octal": [
      2,
    ],
    "no-prototype-builtins": [
      2,
    ],
    "no-redeclare": [
      2,
      {
        "builtinGlobals": true,
      },
    ],
    "no-regex-spaces": [
      2,
    ],
    "no-self-assign": [
      2,
      {
        "props": true,
      },
    ],
    "no-setter-return": [
      2,
    ],
    "no-shadow-restricted-names": [
      2,
      {
        "reportGlobalThis": false,
      },
    ],
    "no-sparse-arrays": [
      2,
    ],
    "no-this-before-super": [
      2,
    ],
    "no-undef": [
      2,
      {
        "typeof": false,
      },
    ],
    "no-unexpected-multiline": [
      2,
    ],
    "no-unreachable": [
      2,
    ],
    "no-unsafe-finally": [
      2,
    ],
    "no-unsafe-negation": [
      2,
      {
        "enforceForOrderingRelations": false,
      },
    ],
    "no-unsafe-optional-chaining": [
      2,
      {
        "disallowArithmeticOperators": false,
      },
    ],
    "no-unused-labels": [
      2,
    ],
    "no-unused-private-class-members": [
      2,
    ],
    "no-unused-vars": [
      2,
    ],
    "no-useless-backreference": [
      2,
    ],
    "no-useless-catch": [
      2,
    ],
    "no-useless-escape": [
      2,
    ],
    "no-with": [
      2,
    ],
    "react-hooks/exhaustive-deps": [
      2,
    ],
    "react-hooks/rules-of-hooks": [
      2,
    ],
    "react/display-name": [
      2,
    ],
    "react/jsx-key": [
      2,
    ],
    "react/jsx-no-comment-textnodes": [
      2,
    ],
    "react/jsx-no-duplicate-props": [
      2,
    ],
    "react/jsx-no-target-blank": [
      2,
    ],
    "react/jsx-no-undef": [
      2,
    ],
    "react/jsx-uses-react": [
      0,
    ],
    "react/jsx-uses-vars": [
      2,
    ],
    "react/no-children-prop": [
      2,
    ],
    "react/no-danger-with-children": [
      2,
    ],
    "react/no-deprecated": [
      2,
    ],
    "react/no-direct-mutation-state": [
      2,
    ],
    "react/no-find-dom-node": [
      2,
    ],
    "react/no-is-mounted": [
      2,
    ],
    "react/no-render-return-value": [
      2,
    ],
    "react/no-string-refs": [
      2,
    ],
    "react/no-unescaped-entities": [
      2,
    ],
    "react/no-unknown-property": [
      2,
    ],
    "react/no-unsafe": [
      0,
    ],
    "react/prop-types": [
      0,
    ],
    "react/react-in-jsx-scope": [
      0,
    ],
    "react/require-render-return": [
      2,
    ],
    "require-yield": [
      2,
    ],
    "simple-import-sort/exports": [
      2,
    ],
    "simple-import-sort/imports": [
      2,
    ],
    "unused-imports/no-unused-imports": [
      2,
    ],
    "use-isnan": [
      2,
      {
        "enforceForIndexOf": false,
        "enforceForSwitchCase": true,
      },
    ],
    "valid-typeof": [
      2,
      {
        "requireStringLiterals": false,
      },
    ],
  },
  "settings": {
    "react": {
      "version": "detect",
    },
  },
}
`;

exports[`スナップショットテスト > index.ts 1`] = `
{
  "language": "@/js",
  "languageOptions": {
    "ecmaVersion": 2025,
    "parser": "typescript-eslint/parser@8.27.0",
    "parserOptions": {
      "ecmaFeatures": {
        "jsx": true,
      },
      "jsxPragma": null,
      "project": true,
    },
    "sourceType": "module",
  },
  "linterOptions": {
    "reportUnusedDisableDirectives": 2,
  },
  "plugins": [
    "@",
    "@typescript-eslint:@typescript-eslint/eslint-plugin@8.27.0",
    "simple-import-sort:eslint-plugin-simple-import-sort@12.1.1",
    "unused-imports:unused-imports",
    "eqeqeq-fix",
    "react",
    "react-hooks:eslint-plugin-react-hooks",
  ],
  "processor": undefined,
  "rules": {
    "@typescript-eslint/await-thenable": [
      2,
    ],
    "@typescript-eslint/ban-ts-comment": [
      2,
      {
        "minimumDescriptionLength": 0,
      },
    ],
    "@typescript-eslint/consistent-type-imports": [
      2,
    ],
    "@typescript-eslint/eqeqeq": [
      0,
    ],
    "@typescript-eslint/method-signature-style": [
      2,
    ],
    "@typescript-eslint/no-array-constructor": [
      2,
    ],
    "@typescript-eslint/no-array-delete": [
      2,
    ],
    "@typescript-eslint/no-base-to-string": [
      2,
    ],
    "@typescript-eslint/no-confusing-void-expression": [
      0,
    ],
    "@typescript-eslint/no-deprecated": [
      2,
    ],
    "@typescript-eslint/no-duplicate-enum-values": [
      2,
    ],
    "@typescript-eslint/no-duplicate-type-constituents": [
      2,
    ],
    "@typescript-eslint/no-dynamic-delete": [
      2,
    ],
    "@typescript-eslint/no-empty-object-type": [
      2,
    ],
    "@typescript-eslint/no-explicit-any": [
      2,
    ],
    "@typescript-eslint/no-extra-non-null-assertion": [
      2,
    ],
    "@typescript-eslint/no-extraneous-class": [
      2,
    ],
    "@typescript-eslint/no-floating-promises": [
      2,
    ],
    "@typescript-eslint/no-for-in-array": [
      2,
    ],
    "@typescript-eslint/no-implied-eval": [
      2,
    ],
    "@typescript-eslint/no-invalid-void-type": [
      2,
    ],
    "@typescript-eslint/no-meaningless-void-operator": [
      2,
    ],
    "@typescript-eslint/no-misused-new": [
      2,
    ],
    "@typescript-eslint/no-misused-promises": [
      2,
      {
        "checksVoidReturn": false,
      },
    ],
    "@typescript-eslint/no-misused-spread": [
      2,
    ],
    "@typescript-eslint/no-mixed-enums": [
      2,
    ],
    "@typescript-eslint/no-namespace": [
      2,
    ],
    "@typescript-eslint/no-non-null-asserted-nullish-coalescing": [
      2,
    ],
    "@typescript-eslint/no-non-null-asserted-optional-chain": [
      2,
    ],
    "@typescript-eslint/no-non-null-assertion": [
      0,
    ],
    "@typescript-eslint/no-redundant-type-constituents": [
      2,
    ],
    "@typescript-eslint/no-require-imports": [
      2,
    ],
    "@typescript-eslint/no-this-alias": [
      2,
    ],
    "@typescript-eslint/no-unnecessary-boolean-literal-compare": [
      2,
    ],
    "@typescript-eslint/no-unnecessary-condition": [
      2,
    ],
    "@typescript-eslint/no-unnecessary-template-expression": [
      2,
    ],
    "@typescript-eslint/no-unnecessary-type-arguments": [
      2,
    ],
    "@typescript-eslint/no-unnecessary-type-assertion": [
      2,
    ],
    "@typescript-eslint/no-unnecessary-type-constraint": [
      2,
    ],
    "@typescript-eslint/no-unnecessary-type-parameters": [
      2,
    ],
    "@typescript-eslint/no-unsafe-argument": [
      2,
    ],
    "@typescript-eslint/no-unsafe-assignment": [
      2,
    ],
    "@typescript-eslint/no-unsafe-call": [
      2,
    ],
    "@typescript-eslint/no-unsafe-declaration-merging": [
      2,
    ],
    "@typescript-eslint/no-unsafe-enum-comparison": [
      2,
    ],
    "@typescript-eslint/no-unsafe-function-type": [
      2,
    ],
    "@typescript-eslint/no-unsafe-member-access": [
      2,
    ],
    "@typescript-eslint/no-unsafe-return": [
      2,
    ],
    "@typescript-eslint/no-unsafe-unary-minus": [
      2,
    ],
    "@typescript-eslint/no-unused-expressions": [
      2,
      {
        "allowShortCircuit": false,
        "allowTaggedTemplates": false,
        "allowTernary": false,
      },
    ],
    "@typescript-eslint/no-unused-vars": [
      0,
    ],
    "@typescript-eslint/no-useless-constructor": [
      2,
    ],
    "@typescript-eslint/no-wrapper-object-types": [
      2,
    ],
    "@typescript-eslint/only-throw-error": [
      2,
    ],
    "@typescript-eslint/prefer-as-const": [
      2,
    ],
    "@typescript-eslint/prefer-literal-enum-member": [
      2,
    ],
    "@typescript-eslint/prefer-namespace-keyword": [
      2,
    ],
    "@typescript-eslint/prefer-promise-reject-errors": [
      2,
    ],
    "@typescript-eslint/prefer-reduce-type-parameter": [
      2,
    ],
    "@typescript-eslint/prefer-return-this-type": [
      2,
    ],
    "@typescript-eslint/related-getter-setter-pairs": [
      2,
    ],
    "@typescript-eslint/require-await": [
      2,
    ],
    "@typescript-eslint/restrict-plus-operands": [
      2,
      {
        "allowAny": false,
        "allowBoolean": false,
        "allowNullish": false,
        "allowNumberAndString": false,
        "allowRegExp": false,
      },
    ],
    "@typescript-eslint/restrict-template-expressions": [
      2,
      {
        "allowNumber": true,
      },
    ],
    "@typescript-eslint/return-await": [
      2,
      "error-handling-correctness-only",
    ],
    "@typescript-eslint/triple-slash-reference": [
      2,
    ],
    "@typescript-eslint/unbound-method": [
      2,
    ],
    "@typescript-eslint/unified-signatures": [
      2,
    ],
    "@typescript-eslint/use-unknown-in-catch-callback-variable": [
      2,
    ],
    "constructor-super": [
      0,
    ],
    "eqeqeq": [
      0,
    ],
    "eqeqeq-fix/eqeqeq": [
      2,
    ],
    "for-direction": [
      2,
    ],
    "getter-return": [
      0,
      {
        "allowImplicit": false,
      },
    ],
    "no-array-constructor": [
      0,
    ],
    "no-async-promise-executor": [
      2,
    ],
    "no-case-declarations": [
      2,
    ],
    "no-class-assign": [
      0,
    ],
    "no-compare-neg-zero": [
      2,
    ],
    "no-cond-assign": [
      2,
      "except-parens",
    ],
    "no-console": [
      2,
      {},
    ],
    "no-const-assign": [
      0,
    ],
    "no-constant-binary-expression": [
      2,
    ],
    "no-constant-condition": [
      2,
      {
        "checkLoops": "allExceptWhileTrue",
      },
    ],
    "no-control-regex": [
      2,
    ],
    "no-debugger": [
      2,
    ],
    "no-delete-var": [
      2,
    ],
    "no-dupe-args": [
      0,
    ],
    "no-dupe-class-members": [
      0,
    ],
    "no-dupe-else-if": [
      2,
    ],
    "no-dupe-keys": [
      0,
    ],
    "no-duplicate-case": [
      2,
    ],
    "no-empty": [
      2,
      {
        "allowEmptyCatch": false,
      },
    ],
    "no-empty-character-class": [
      2,
    ],
    "no-empty-pattern": [
      2,
      {
        "allowObjectPatternsAsParameters": false,
      },
    ],
    "no-empty-static-block": [
      2,
    ],
    "no-ex-assign": [
      2,
    ],
    "no-extra-boolean-cast": [
      2,
      {},
    ],
    "no-fallthrough": [
      2,
      {
        "allowEmptyCase": false,
        "reportUnusedFallthroughComment": false,
      },
    ],
    "no-func-assign": [
      0,
    ],
    "no-global-assign": [
      2,
      {
        "exceptions": [],
      },
    ],
    "no-implied-eval": [
      0,
    ],
    "no-import-assign": [
      0,
    ],
    "no-invalid-regexp": [
      2,
      {},
    ],
    "no-irregular-whitespace": [
      2,
      {
        "skipComments": false,
        "skipJSXText": false,
        "skipRegExps": false,
        "skipStrings": true,
        "skipTemplates": false,
      },
    ],
    "no-loss-of-precision": [
      2,
    ],
    "no-misleading-character-class": [
      2,
    ],
    "no-new-native-nonconstructor": [
      0,
    ],
    "no-new-symbol": [
      0,
    ],
    "no-nonoctal-decimal-escape": [
      2,
    ],
    "no-obj-calls": [
      0,
    ],
    "no-octal": [
      2,
    ],
    "no-prototype-builtins": [
      2,
    ],
    "no-redeclare": [
      0,
      {
        "builtinGlobals": true,
      },
    ],
    "no-regex-spaces": [
      2,
    ],
    "no-return-await": [
      0,
    ],
    "no-self-assign": [
      2,
      {
        "props": true,
      },
    ],
    "no-setter-return": [
      0,
    ],
    "no-shadow-restricted-names": [
      2,
      {
        "reportGlobalThis": false,
      },
    ],
    "no-sparse-arrays": [
      2,
    ],
    "no-this-before-super": [
      0,
    ],
    "no-throw-literal": [
      0,
    ],
    "no-undef": [
      0,
      {
        "typeof": false,
      },
    ],
    "no-unexpected-multiline": [
      2,
    ],
    "no-unreachable": [
      0,
    ],
    "no-unsafe-finally": [
      2,
    ],
    "no-unsafe-negation": [
      0,
      {
        "enforceForOrderingRelations": false,
      },
    ],
    "no-unsafe-optional-chaining": [
      2,
      {
        "disallowArithmeticOperators": false,
      },
    ],
    "no-unused-expressions": [
      0,
      {
        "allowShortCircuit": false,
        "allowTaggedTemplates": false,
        "allowTernary": false,
        "enforceForJSX": false,
        "ignoreDirectives": false,
      },
    ],
    "no-unused-labels": [
      2,
    ],
    "no-unused-private-class-members": [
      2,
    ],
    "no-unused-vars": [
      0,
    ],
    "no-useless-backreference": [
      2,
    ],
    "no-useless-catch": [
      2,
    ],
    "no-useless-constructor": [
      0,
    ],
    "no-useless-escape": [
      2,
    ],
    "no-var": [
      2,
    ],
    "no-with": [
      2,
    ],
    "prefer-const": [
      2,
      {
        "destructuring": "any",
        "ignoreReadBeforeAssign": false,
      },
    ],
    "prefer-promise-reject-errors": [
      0,
      {
        "allowEmptyReject": false,
      },
    ],
    "prefer-rest-params": [
      2,
    ],
    "prefer-spread": [
      2,
    ],
    "react-hooks/exhaustive-deps": [
      2,
    ],
    "react-hooks/rules-of-hooks": [
      2,
    ],
    "react/display-name": [
      2,
    ],
    "react/jsx-key": [
      2,
    ],
    "react/jsx-no-comment-textnodes": [
      2,
    ],
    "react/jsx-no-duplicate-props": [
      2,
    ],
    "react/jsx-no-target-blank": [
      2,
    ],
    "react/jsx-no-undef": [
      2,
    ],
    "react/jsx-uses-react": [
      0,
    ],
    "react/jsx-uses-vars": [
      2,
    ],
    "react/no-children-prop": [
      2,
    ],
    "react/no-danger-with-children": [
      2,
    ],
    "react/no-deprecated": [
      2,
    ],
    "react/no-direct-mutation-state": [
      2,
    ],
    "react/no-find-dom-node": [
      2,
    ],
    "react/no-is-mounted": [
      2,
    ],
    "react/no-render-return-value": [
      2,
    ],
    "react/no-string-refs": [
      2,
    ],
    "react/no-unescaped-entities": [
      2,
    ],
    "react/no-unknown-property": [
      2,
    ],
    "react/no-unsafe": [
      0,
    ],
    "react/prop-types": [
      0,
    ],
    "react/react-in-jsx-scope": [
      0,
    ],
    "react/require-render-return": [
      2,
    ],
    "require-await": [
      0,
    ],
    "require-yield": [
      2,
    ],
    "simple-import-sort/exports": [
      2,
    ],
    "simple-import-sort/imports": [
      2,
    ],
    "unused-imports/no-unused-imports": [
      2,
    ],
    "use-isnan": [
      2,
      {
        "enforceForIndexOf": false,
        "enforceForSwitchCase": true,
      },
    ],
    "valid-typeof": [
      2,
      {
        "requireStringLiterals": false,
      },
    ],
  },
  "settings": {
    "react": {
      "version": "detect",
    },
  },
}
`;
